lib/puppet/provider/package/puppet_gem.rb

```diff
@@ -14,4 +14,13 @@
   else
     commands :gemcmd => "/opt/puppetlabs/puppet/bin/gem"
   end
+
+  def uninstall
+    begin
+      super
+    ensure
```

Comment:
Why's this in an ensure block?

Reply:
I thought it might be wise or at least wouldn't hurt to invalidate the cache even if there was some sort of error during the uninstall.

Reply:
One potential issue is if the uninstall raises and the call to `gem_source.clear_paths` also raises, then we lose the original exception. For example given:

```ruby
begin
  raise "from inside"
ensure
  raise "from ensure"
end
```

The caller will receive the "from ensure" exception instead of the original:

```
$ bundle exec ruby raise.rb
Traceback (most recent call last):
	1: from raise.rb:4:in `<main>'
raise.rb:4:in `ensure in <main>': from ensure (RuntimeError)
```

So I'm thinking it's better to only clear the rubygems cache if we successfully uninstall the gem.

Reply:
OK.  Updated to remove use of `ensure`